From a45bd0d9682db986191b075e64bcacd038af5993 Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@redhat.com>
Date: Thu, 1 Sep 2011 10:45:00 +0200
Subject: [PATCH] Don't treat tls_offset == 1 as forced dynamic

* include/link.h (FORCED_DYNAMIC_TLS_OFFSET) [NO_TLS_OFFSET == 0]:
Change to -1.

---
 ChangeLog      |    5 +++++
 include/link.h |    2 +-
 2 files changed, 6 insertions(+), 1 deletions(-)

--- a/include/link.h
+++ b/include/link.h
@@ -289,7 +289,7 @@ struct link_map
 #endif
 #ifndef FORCED_DYNAMIC_TLS_OFFSET
 # if NO_TLS_OFFSET == 0
-#  define FORCED_DYNAMIC_TLS_OFFSET 1
+#  define FORCED_DYNAMIC_TLS_OFFSET -1
 # elif NO_TLS_OFFSET == -1
 #  define FORCED_DYNAMIC_TLS_OFFSET -2
 # else
